scripts/resign_analysis/resign_analysis.py

```diff
@@ -17,156 +17,68 @@
-            stats[gs.winner].resigned_game_len_sum += gs.resign_movenum
-        stats[gs.winner].game_len_sum += gs.total_moves
-    stats[2].calcOverall(stats[0], stats[1])
-    print("Black won %d/%d (%0.2f%%)" % (
-        stats[0].num_games,
-        stats[0].num_games+stats[1].num_games,
-        100.0 * stats[0].num_games / (stats[0].num_games+stats[1].num_games)))
-    for winner in (0,1,2):
-        if winner==0:
-            print("Of games Black won:")
-        elif winner==1:
-            print("Of games White won:")
-        else:
-            print("Of all games:")
-        if stats[winner].num_games == 0:
-            print("    No games to report")
-            continue
-        avg_len = 1.0*stats[winner].game_len_sum/stats[winner].num_games
-        resigned_avg_len = 1.0*stats[winner].resigned_game_len_sum/stats[winner].num_games
-        avg_reduction = 1.0*(avg_len-resigned_avg_len)/avg_len
-        print("    Incorrect uct resigns = %d/%d (%0.2f%%)" % (
-            stats[winner].incorrect_resign_count,
-            stats[winner].num_games,
-            100.0 * stats[winner].incorrect_resign_count / stats[winner].num_games))
-        print("    Correct uct resigns = %d/%d (%0.2f%%)" % (
-            stats[winner].correct_resign_count,
-            stats[winner].num_games,
-            100.0 * stats[winner].correct_resign_count / stats[winner].num_games))
-        print("    No Resign = %d/%d (%0.2f%%)" % (
-            stats[winner].no_resign_count,
-            stats[winner].num_games,
-            100.0 * stats[winner].no_resign_count / stats[winner].num_games))
-        print("    Average game length = %d. Average game length with resigns = %d (%0.2f%% reduction)" % (
-            avg_len, resigned_avg_len, avg_reduction*100))
+        #print("debug filename: {}".format(filename))
+        plynum = 0
+        who_resigned = None
+        resign_plynum = None
+        for line in open(filename).readlines():
+            m = re.match("^Score: (\S+)", line)
+            if m:
+                (score,) = m.groups()
+                score = float(score)
+                #print("debug Score", score)
+            m = re.match("^info string stm\s+(\S+) winrate\s+(\S+)%", line)
+            if m:
+                plynum += 1
+                total_plys += 1
+                if who_resigned:
+                    # We already found who resigned.
+                    # continue after just counting plys
+                    continue
+                resign_plys += 1
+                (stm, winrate) = m.groups()
+                winrate = float(winrate)
+                if winrate < resignrate:
+                    who_resigned = stm
+                    resign_plynum = plynum
+                    #print("debug stm, winrate, plynum", stm, winrate, plynum)
+        #print("debug who_resigned {} resign_playnum {} total_plynum {}".format(who_resigned, resign_plynum, plynum))
+        if ((score == -1 and who_resigned == "Black") or
```

Comment:
Shouldn't this be score != 1 and score != -1 on the next line?
Converting a resignation in to a draw is also an incorrect resignation?

Reply:
It's only half as bad, so I think it should only have half as much affect on increasing FP rate.